src/urllib3/connection.py

```diff
@@ -151,6 +155,12 @@ def __init__(
         self.socket_options = socket_options
         self.proxy = proxy
         self.proxy_config = proxy_config
+        if idle_timeout:
+            if isinstance(idle_timeout, datetime.timedelta):
+                self.idle_timeout = idle_timeout
+            else:
+                self.idle_timeout = datetime.timedelta(seconds=idle_timeout)
+        self.last_activity = datetime.datetime.now()
```

Comment:
Would it be preferable to use monotonic time for this? `datetime` introduces the complexity of timezones and clock jumps, and keeping it all as floats keeps typing consistent with the other timeouts. eg.
```suggestion
        if idle_timeout:
            if isinstance(idle_timeout, datetime.timedelta):
                self.idle_timeout = idle_timeout.total_seconds()
            else:
                self.idle_timeout = idle_timeout
        self.last_activity = time.monotonic()
```